backend_jobs/ingestion_pipeline/pipeline_lib/database_functions.py

```diff
@@ -0,0 +1,143 @@
+        else:
+            #doc not found- image has not been ingested already
+            add_document(element, provider, doc_ref)
+        #Adding a doc to the sub collection (pipelinerun) in the image collection
+        upload_sub_collection(element, provider, job_name, sub_collection_doc_ref)
+
+def add_document(element, provider, doc_ref):
+    geo_point_coordinates=cloud_firestore.GeoPoint(
+        float(element.coordinates['latitude']),
+        float(element.coordinates['longitude']))
+    doc_ref.set({
+        u'url': element.url,
+        u'ingestedProviders':[provider.provider_id],
+        u'ingestedRuns':[1],
+        u'coordinates': geo_point_coordinates,
+        u'dateIngested': datetime.now(),
+        u'dateShot': element.date_shot,
+        u'dateFields':get_date_fields(element.date_shot),
+        u'geoHashes': get_geo_hashes_map(element.coordinates),
+        u'imageAttributes':{
+            u'format': element.format,
+            u'resolution':element.resolution},
+        u'attribution': element.attribution,
+        u'random': random.randint(1,101),
+        u'visibility': provider.visibility.value,
+    })
+
+def update_document(provider,doc, doc_ref,job_name):
+    ingested_runs = doc.to_dict()[u'ingestedRuns']
+    ingested_runs.append(job_name)
+    ingested_providers = doc.to_dict()[u'ingestedProviders']
+    if provider.provider_id not in ingested_providers:
+        ingested_providers.append(provider.provider_id)
+    doc_ref.update({
+        u'ingestedRuns':ingested_runs,
+        u'ingestedProviders': ingested_providers,
+        u'visibility': max(provider.visibility, doc.to_dict()[u'visibility']),
+    })
+
+def upload_sub_collection(element, provider,job_name, sub_collection_doc_ref):
+    sub_collection_doc_ref.set({
+        u'coordinates': element.coordinates,
+        u'provider_ID':provider.provider_id,
+        u'provider_version': provider.provider_version,
+        u'provider_type':element.provider_type.value,
+        u'provider_visibility': provider.visibility.value,
+        u'pipeline_run': job_name,
+        u'geoHashes': get_geo_hashes_map(element.coordinates),
+    })
+
+
+def get_geo_hashes_map(coordinates):
+    """
+    This function, given a coordinates (lat,long), calculates the geohash
+    and builds a map containing a geohash in different lengths.
+    """
+    geo_hashes_map={}
+    geohash=geohash2.encode(coordinates['latitude'],coordinates['longitude'])
+    for i in range(1,10,1):
+        geo_hashes_map['hash'+str(i)]= geohash[0:i]
+    return geo_hashes_map
+
+def get_date_fields(date):
+    """
+    This function converts a datetime object to a map object contaning the date
+    """
+    year=date.year
+    month=date.month
+    day=date.day
+    date_fields={'year':year,'month':month,'day':day}
+    return date_fields
+
+def get_doc_by_id(image_id):
+    """
+    This function given a id returns a the doc in IMAGES_COLLECTION
```

Comment:
doc

Reply:
did not need this function so removed it.